lib/dotcom/system_status/groups.ex

```diff
@@ -0,0 +1,337 @@
+    alerts
+    |> Enum.filter(fn %Alert{informed_entity: informed_entity} ->
+      informed_entity
+      |> Enum.any?(fn
+        %{route: ^line_id} -> true
+        %{} -> false
+      end)
+    end)
+  end
+
+  # Maps a list of alerts to a list of statuses, where a status is a
+  # simple structure with a route, a description, and a few additional
+  # fields that determine how it will render in the frontend.
+  defp alerts_to_statuses(alerts, time)
+
+  # If there are no alerts, then we want a single status indicating
+  # "Normal Service".
+  defp alerts_to_statuses([], _time) do
+    [%{description: "Normal Service", time: nil}]
+  end
+
+  # If there are alerts, then create a starting list of statuses that
+  # maps one-to-one with the alerts provided.
+  defp alerts_to_statuses(alerts, time) do
+    alerts
+    |> Enum.map(fn alert ->
+      alert_to_status(alert, time)
+    end)
+  end
+
+  # Translates an alert to a status:
+  #  - The effect is humanized into a description for the status.
+  #  - If the alert's already active, `time` is set to `nil`.
+  #  - If the alert is in the future, `time` is set to the alert's
+  #    start time
+  defp alert_to_status(alert, time) do
+    description =
+      case alert.effect do
+        :delay -> "Delays"
+        :shuttle -> "Shuttle Buses"
+        :station_closure -> "Station Closure"
+        :suspension -> "Suspension"
+      end
+
+    time = future_start_time(alert.active_period, time)
+
+    %{description: description, time: time}
+  end
+
+  # - If the active period is in the future, returns its start_time.
+  # - If the active period indicates that the alert is currently active, returns nil.
+  # - Raises an error if the alert is completely in the past.
+  defp future_start_time(
+         [{start_time, _end_time} = first_active_period | more_active_periods],
+         time
+       ) do
+    cond do
+      ends_before?(first_active_period, time) -> future_start_time(more_active_periods, time)
+      starts_before?(first_active_period, time) -> nil
+      true -> start_time
+    end
+  end
+
+  # Returns true if the active period ends before the time given. An
+  # end-time of false indicates an indefinite active period, which
+  # never ends.
+  defp ends_before?({_start_time, nil}, _time), do: false
+  defp ends_before?({_start_time, end_time}, time), do: Timex.before?(end_time, time)
+
+  # Returns true if the active period starts before the time given.
+  defp starts_before?({start_time, _end_time}, time), do: Timex.before?(start_time, time)
+
+  # Combines statuses that have the same active time and description
+  # into a single pluralized status (e.g. "Station Closures" instead
+  # of "Station Closure").
+  defp consolidate_duplicate_descriptions(statuses) do
+    statuses
+    |> Enum.group_by(fn %{time: time, description: description} -> {time, description} end)
+    |> Enum.map(fn
+      {_, [status]} -> status
+      {_, [status | _]} -> pluralize_description(status)
```

Comment:
Looks like `Inflex.pluralize/1` is smart enough to ignore existing plurals by the way!

```
iex(2)> ["Delays", "Shuttle Buses", "Station Closure", "Suspension"] |> Enum.map(&Inflex.pluralize/1)
["Delays", "Shuttle Buses", "Station Closures", "Suspensions"]
```

Reply:
Touché!